scriptmodules/emulators/mame.sh

```diff
@@ -0,0 +1,152 @@
+#!/usr/bin/env bash
+
+# This file is part of The RetroPie Project
+#
+# The RetroPie Project is the legal property of its developers, whose names are
+# too numerous to list here. Please refer to the COPYRIGHT.md file distributed with this source.
+#
+# See the LICENSE.md file at the top-level directory of this distribution and
+# at https://raw.githubusercontent.com/RetroPie/RetroPie-Setup/master/LICENSE.md
+#
+
+rp_module_id="mame"
+rp_module_desc="MAME emulator"
+rp_module_help="ROM Extensions: .zip .7z\n\nCopy your MAME roms to either $romdir/mame or\n$romdir/arcade"
+rp_module_licence="GPL2 https://github.com/mamedev/mame/blob/master/COPYING"
+rp_module_section="exp"
+rp_module_flags="!mali !armv6"
+
+function _latest_ver_mame() {
+    wget -qO- https://api.github.com/repos/mamedev/mame/releases/latest | grep -m 1 tag_name | cut -d\" -f4
+}
+
+function _get_binary_name_mame() {
+    # The MAME executable on 64-bit systems is called mame64 instead of mame. Rename it back to mame.
+    if isPlatform "64bit"; then
+        echo 'mame64'
+    else
+        echo 'mame'
+    fi
+}
+
+function depends_mame() {
+    if compareVersions $__gcc_version lt 6.0.0; then
+        md_ret_errors+=("Sorry, you need an OS with gcc 6.0 or newer to compile mame")
+        return 1
+    fi
+
+    # Additional libraries required for running
+    getDepends libsdl2-ttf-2.0-0
+
+    # Additional libraries required for compilation
+    # Note: libxi-dev is requires as of v0.210, because of flag changes for XInput
+    getDepends libfontconfig1-dev qt5-default libsdl2-ttf-dev libxinerama-dev libxi-dev
+}
+
+function sources_mame() {
+    gitPullOrClone "$md_build" https://github.com/mamedev/mame.git "$(_latest_ver_mame)"
+}
+
+function build_mame() {
+    # More memory is required for x86 platforms
+    if isPlatform "x86"; then
+        rpSwap on 8192
+    else
+        rpSwap on 4096
+    fi
+
+    # Compile MAME
+    local params=(NOWERROR=1 ARCHOPTS=-U_FORTIFY_SOURCE)
+    make "${params[@]}"
+
+    strip "$(_get_binary_name_${md_id})"
+
+    rpSwap off
+    md_ret_require="$md_build/$(_get_binary_name_${md_id})"
```

Comment:
I'd get the name once into a var - avoids duplicate calls and less duplication.

Reply:
Do you mean once per function where it is used, or a global variable?

Reply:
In a function when it's used more than once. Not a global.

Reply:
Changed to use a variable instead of calling the function multiple times, when applicable.